extension/data/background/handlers/modqueue.js

```diff
@@ -1,9 +1,40 @@
 'use strict';
 
 const MODQUEUE_CACHE_TTL = 30;
+const MODQUEUE_CACHE_NAME = 'tb-modqueue-cache';
 
-// Object containing the queue cached data per subreddit
-const queueCache = new Map();
+/**
+ * Sets the modqueue cache for a given subreddit.
+ * @param subreddit string subreddit
+ * @param cacheObject object containing the cache
+ */
+function setQueueCache (subreddit, cacheObject) {
+    return new Promise(resolve => {
+        browser.storage.local.get({[MODQUEUE_CACHE_NAME]: {}}).then(result => {
+            result[subreddit] = cacheObject;
+            browser.storage.local.set({
+                [`${MODQUEUE_CACHE_NAME}`]: result,
+            }).then(resolve);
+        });
+    });
+}
+
+/**
+ * Returns the modqueue cache for a given subreddit.
+ * @param subreddit subreddit
+ * @returns {promise<object>}
+ */
+function getQueueCache (subreddit) {
+    return new Promise(resolve => {
+        browser.storage.local.get({[MODQUEUE_CACHE_NAME]: {}}).then(result => {
+            if (Object.prototype.hasOwnProperty.call(result[MODQUEUE_CACHE_NAME], subreddit)) {
+                resolve(result[MODQUEUE_CACHE_NAME][subreddit]);
+            } else {
+                resolve({});
+            }
+        });
+    });
+}
```

Comment:
Similar to above, could be cleaner as an async function

```suggestion
async function getQueueCache (subreddit) {
    const result = await browser.storage.local.get({[MODQUEUE_CACHE_NAME]: {}});
    if (Object.prototype.hasOwnProperty.call(result[MODQUEUE_CACHE_NAME], subreddit)) {
        return result[MODQUEUE_CACHE_NAME][subreddit];
    }
    return {};
}
```